### PR TITLE
Improve the paging performance a bit

### PR DESCRIPTION
**Pull request by mihasic**
Skip MaxCount+1 message from being processed

### CODE REVIEW

**Comment by damianh** [comment]
Thanks for the PR @mihasic . Can you comment in the `Files Changed` the specific area that was modified? I think indent change has mark too many lines as changed.

File: src/SqlStreamStore.MsSql/MsSqlStreamStore.ReadAll.cs [hunk 1]
```diff
@@ -46,38 +46,45 @@ public partial class MsSqlStreamStore
...
-                        var streamVersion = reader.GetInt32(1);
-                        ordinal = reader.GetInt64(2);
-                        var eventId = reader.GetGuid(3);
-                        var created = reader.GetDateTime(4);
-                        var type = reader.GetString(5);
-                        var jsonMetadata = reader.GetString(6);
-
-                        Func<CancellationToken, Task<string>> getJsonData;
-                        if (prefetch)
+                        if(messages.Count == maxCount)
```
Comment (mihasic) [inline src/SqlStreamStore.MsSql/MsSqlStreamStore.ReadAll.cs]: Wrapped internals of loop in this if... else. Basically, last n+1 message is not fully read, but replaced with default(...)

File: src/SqlStreamStore.MsSql/MsSqlStreamStore.ReadStream.cs [hunk 2]
```diff
@@ -115,34 +115,42 @@ public partial class MsSqlStreamStore
...
-                        var streamVersion1 = reader.GetInt32(0);
-                        var ordinal = reader.GetInt64(1);
-                        var eventId = reader.GetGuid(2);
-                        var created = reader.GetDateTime(3);
-                        var type = reader.GetString(4);
-                        var jsonMetadata = reader.GetString(5);
-
-                        Func<CancellationToken, Task<string>> getJsonData;
-                        if(prefetch)
+                        if(messages.Count == count)
```
Comment (mihasic) [inline src/SqlStreamStore.MsSql/MsSqlStreamStore.ReadStream.cs]: Same here

Comment (damianh) [inline src/SqlStreamStore.MsSql/MsSqlStreamStore.ReadStream.cs]: Ok, cheers!

**Pull request merged**